File: FizzBuzz/TDD/kotlin/src/main/kotlin/FizzBuzz.kt
object FizzBuzz {

	fun stringFor(input: Any): String
	{
		val result = process(input)
		return print(result)
	}

	private fun process(input: Any): List<String>
	{
		return ProcessJob.process(input)
	}

	private fun print(value: List<String>): String
	{
		val printers = PrintersProvider.all()
		val lastPrinter = printers.last()
		return lastPrinter(value)
	}
}

File: FizzBuzz/TDD/Scala/src/main/scala/FizzBuzz.scala
object FizzBuzz {

  def stringFor(input: Any): Any = {
    val result = process(input)
    print(result)
  }

  private def process(input: Any): Any = {
    ProcessJob.process(input)
  }

  private def print(value: Any): Any = {
    PrinterJob.print(value)
  }
}
